pkg/framework/test/examples_api_server_special_cert_dir_manager_test.go

```diff
@@ -0,0 +1,38 @@
+package test_test
+
+import (
+	"fmt"
+
+	"io/ioutil"
+	"os"
+
+	. "k8s.io/kubectl/pkg/framework/test"
+)
+
+func ExampleAPIServer_credHubCertDirManager() {
+	apiServer := &APIServer{
+		CertDirManager: NewCredHubCertDirManager(),
```

Comment:
Again, can't we just provide a path? If the path is present, we assume it's user managed. If the path is not, then it's apiserver managed. Then we can hide a *ton* of the directory management thing (much simpler to learn). If I provide my directory, I don't think I want the apiserver object to mess up with it too much (maybe I want to re-use it). And I shouldn't have to create a new class for that, or learn about "TempDirManager", "Remover" and all.

Reply:
And this example then becomes unnecessary